### PR TITLE
Add list of VM sizes to AKS UI proxy 

### PR DESCRIPTION
**Pull request by mjura**
Issue: https://github.com/rancher/rancher/issues/32587

### CODE REVIEW

File: pkg/api/norman/customization/aks/listers.go [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+	if err != nil {
+		return nil, http.StatusInternalServerError, err
+	}
+
+	vmMachineSizeList, err := virtualMachine.List(ctx, cap.ResourceLocation)
+	if err != nil {
+		return nil, http.StatusBadRequest, fmt.Errorf("failed to get VM sizes: %v", err)
+	}
+
+	var vmSizes []string
```
Comment (thedadams) [inline pkg/api/norman/customization/aks/listers.go:214]:
```suggestion
	vmSizes := make([]string, 0, len(*vmMachineSizeList.Value))
```

**Review by thedadams — CHANGES_REQUESTED** [review]
One change, but #32454 should be merged first.

File: pkg/api/norman/customization/aks/listers.go (hunk 1 above)
Comment (mjura) [reply pkg/api/norman/customization/aks/listers.go]: changed

**Review by cmurphy — COMMENTED** [review]
lgtm once https://github.com/rancher/rancher/pull/32454 is merged

File: pkg/api/norman/customization/aks/handler.go [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+	TenantID         string `json:"tenantId"`
+	SubscriptionID   string `json:"subscriptionId"`
+	ClientID         string `json:"clientId"`
+	ClientSecret     string `json:"clientSecret"`
+	ResourceLocation string `json:"region"`
+}
+
+// AKS handler lists available resources in Azure API
+type handler struct {
+	Action        string
```
Comment (thedadams) [inline pkg/api/norman/customization/aks/handler.go:43]: Is this needed?

File: go.sum [hunk 3]
```diff
@@ -51,32 +51,48 @@ github.com/Azure/go-autorest v14.2.0+incompatible/go.mod h1:r+4oMnoxhatjLLJ6zxSW
 github.com/Azure/go-autorest/autorest v0.9.0/go.mod h1:xyHB1BMZT0cuDHU7I0+g046+BFDTQ8rEZB0s4Yfa6bI=
 github.com/Azure/go-autorest/autorest v0.9.3/go.mod h1:GsRuLYvwzLjjjRoWEIyMUaYq8GNUx2nRB378IPt/1p0=
 github.com/Azure/go-autorest/autorest v0.11.12/go.mod h1:eipySxLmqSyC5s5k1CLupqet0PSENBEDP93LQ9a8QYw=
+github.com/Azure/go-autorest/autorest v0.11.12/go.mod h1:eipySxLmqSyC5s5k1CLupqet0PSENBEDP93LQ9a8QYw=
```
Comment (thedadams) [inline go.sum:54]: `go.sum` changes without `go.mod` changes?

**Review by thedadams — COMMENTED** [review]
Assuming most of this is identical to https://github.com/rancher/rancher/pull/32454

File: go.sum (hunk 3 above)
Comment (mjura) [reply go.sum]: I fixed it

File: pkg/api/norman/customization/aks/handler.go (hunk 2 above)
Comment (mjura) [reply pkg/api/norman/customization/aks/handler.go]: fixed

**Review by cmurphy — DISMISSED** [review]

**Review by cbron — APPROVED** [review]
lgtm but wait for donnie

**Comment by cmurphy** [comment]
Rebased

**Review by cmurphy — APPROVED** [review]

**Review by thedadams — APPROVED** [review]

**Review by cbron — APPROVED** [review]

**Pull request merged**